src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -896,6 +901,24 @@ public GcpBigQueryDatasetResource createControlledBigQueryDataset(
         "Error creating controlled BigQuery dataset in the workspace.");
   }
 
+  /** Used by tests only. */
```

Comment:
i don't understand. Could you add a DataSource.java here https://github.com/DataBiosphere/terra-cli/tree/2cd3d1cd9cc4bd30516e965ada55590816d4544d/src/main/java/bio/terra/cli/command/resource/addref so this method is used and your test is actually testing a real behavior.